src/sonic-config-engine/minigraph.py

```diff
@@ -1717,6 +1718,19 @@ def parse_xml(filename, platform=None, port_config_file=None, asic_name=None, hw
             results['LOOPBACK_INTERFACE'][host_lo_intf[0]] = {}
 
     results['MGMT_VRF_CONFIG'] = mvrf
+    # Update SNMP_AGENT_ADDRESS_CONFIG with Management IP and Loopback0 IP
+    # if available.
+    if not is_multi_asic() and asic_name is None:
+        results['SNMP_AGENT_ADDRESS_CONFIG'] = {}
+        port = 161
+        for mgmt_if in results['MGMT_INTERFACE'].keys():
+            snmp_key = mgmt_if[1].split('/')[0] + '|' + str(port)  + '|'
+            results['SNMP_AGENT_ADDRESS_CONFIG'][snmp_key] = {}
+        # Add Loopback0 IP as agent address for single asic
+        for loip in results['LOOPBACK_INTERFACE']:
+            if len(loip) == 2 and loip[0] == 'Loopback0':
```

Comment:
condition `len(loip)` seems very strict. This condition will break if an additional loopback interface is introduced in the minigraph.

Reply:
@arlakshm thanks for reviewing, the len(loip) is to make sure that the loip inside LOOPBACK_INTERFACE is actually <name of interface>|<IP address>.
For example, we have 
```
       "LOOPBACK_INTERFACE": {
        "Loopback0": {}, -- This does not contain the IP address, so we don't need to parse this.
        "Loopback0|10.1.0.32/32": {}, -- here the len(loip) will be 2 and we will be able to get the IP address.
        
(Pdb) loip
('Loopback0', '10.1.0.32/32')
(Pdb) len(loip)
2
```